### DISCUSSION TITLE
[Proposal] Target typed generic type parameter

### DISCUSSION THREAD

**Discussion by hillin** [dotnet/csharplang#8666] [Language Ideas] [answered]
This is kind of a generalization of [target typed new](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new).
If a generic method takes exactly one generic parameter, and returns a value of that type, the generic parameter could be omitted.

Example:

```csharp
public MyService MyService => Inject<MyService>();
// and a whole lot of other injections

private T Inject<T>() => ServiceProvider.GetRequiredService<T>();
```

could benefit from this feature:
```csharp
public MyService MyService => Inject();
// and a whole lot of other injections

private T Inject<T>() => ServiceProvider.GetRequiredService();
```

**Comment by HaloFour** [comment, marked as answer]
See: #92
